### PR TITLE
tetragon: Unflake the TestGeneratedExecEvents test

### PR DESCRIPTION
**Pull request by olsajiri**
On slow server the test might fails to start exporter fast enough making the test fail, because events did not get stored.

Moving the tetragonExecuteCtx into separate process and adding extra 5 tries on checking up on the exported file.

### CODE REVIEW

**Review by tpapagian — APPROVED** [review]

File: cmd/tetragon/main_test.go [hunk 1]
```diff
@@ -52,15 +51,39 @@ func TestGeneratedExecEvents(t *testing.T) {
...
 		ec.NewProcessExecChecker("").WithProcess(
 			ec.NewProcessChecker().WithPid(1)),
 	)
 
-	err = jsonchecker.JsonTestCheck(t, checker)
+	// Try it 5 times and make sure exporter is up and processed all data
+	// in case it lags for some reason like slow CI server.
+	cnt := 0
+	for cnt < 5 {
+		if err = jsonchecker.JsonTestCheck(t, checker); err == nil {
```
Comment (mtardy) [inline cmd/tetragon/main_test.go:80]:
```suggestion
		if jsonchecker.JsonTestCheck(t, checker) == nil {
```
FYI the error you will check on line 88 will not be this one as this one is scoped only for this block.

**Review by mtardy — COMMENTED** [review]
lgtm except maybe something with `assert.NoError(t, err)` that might not be correct.

File: cmd/tetragon/main_test.go (hunk 1 above)
Comment (olsajiri) [reply cmd/tetragon/main_test.go:80]: @mtardy hum, err is created on line 43, line 80 won't create new scope ('=' vs ':=' ) .. what do I miss?

Comment (mtardy) [reply cmd/tetragon/main_test.go:80]: ah sorry I missed it somehow I read `:=` because I always use that in if statement... you are right!

**Review by mtardy — APPROVED** [review]

**Pull request merged**